Projects/DbCLI/SearchDb.py

```diff
@@ -510,114 +515,117 @@ def poolFromCurs(curs, similarityMethod):
                   help='read compressed mols from the database')
 
-parser.add_option('--pharm2DTableName', default='pharm2dfps',
+parser.add_argument('--pharm2DTableName', default='pharm2dfps',
                   help='name of the Pharm2D fingerprints table')
-parser.add_option('--fdefFile', '--fdef',
+parser.add_argument('--fdefFile', '--fdef',
                   default=os.path.join(RDConfig.RDDataDir, 'Novartis1.fdef'),
                   help='provide the name of the fdef file to use for 2d pharmacophores')
-parser.add_option('--gobbi2DTableName', default='gobbi2dfps',
+parser.add_argument('--gobbi2DTableName', default='gobbi2dfps',
                   help='name of the Gobbi2D fingerprints table')
 
-parser.add_option(
+parser.add_argument(
   '--similarityType', '--simType', '--sim', default='RDK', choices=supportedSimilarityMethods,
-  help='Choose the type of similarity to use, possible values: RDK, AtomPairs, TopologicalTorsions, Pharm2D, Gobbi2D, Avalon, Morgan. The default is %default')
-parser.add_option('--morganFpDbName', default='Fingerprints.sqlt',
+  help='Choose the type of similarity to use, possible values: RDK, AtomPairs, TopologicalTorsions, Pharm2D, Gobbi2D, Avalon, Morgan. The default is %(default)s')
+parser.add_argument('--morganFpDbName', default='Fingerprints.sqlt',
                   help='name of the morgan fingerprints database')
-parser.add_option('--morganFpTableName', default='morganfps',
+parser.add_argument('--morganFpTableName', default='morganfps',
                   help='name of the morgan fingerprints table')
-parser.add_option('--morganFpColName', default='morganfp',
+parser.add_argument('--morganFpColName', default='morganfp',
                   help='name of the morgan fingerprint column')
 
-parser.add_option(
+parser.add_argument(
   '--similarityMetric', '--simMetric', '--metric', default='',
   choices=('tanimoto', 'dice', 'tversky', ''),
   help='Choose the type of similarity to use, possible values: tanimoto, dice, tversky. The default is determined by the fingerprint type')
-parser.add_option('--tverskyA', default=0.5, type='float', help='Tversky A value')
-parser.add_option('--tverskyB', default=0.5, type='float', help='Tversky B value')
-parser.add_option(
-  '--simThresh', default=-1, type='float',
+parser.add_argument('--tverskyA', default=0.5, type=float, help='Tversky A value')
+parser.add_argument('--tverskyB', default=0.5, type=float, help='Tversky B value')
+parser.add_argument(
+  '--simThresh', default=-1, type=float,
   help='threshold to use for similarity searching. If provided, this supersedes the topN argument')
 
 if __name__ == '__main__':
   import sys, getopt, time
 
-  options, args = parser.parse_args()
-  if len(args) != 1 and not (options.smilesQuery or options.smartsQuery or options.propQuery):
+  options = parser.parse_args()
+  if len(options) != 1 and not (options.smilesQuery or options.smartsQuery or options.propQuery):
```

Comment:
This was the positional argument for a filename. You can see from the code that it is optional provided one of the queries is set. This means the positional argument must be optional and has a default value of None.

Reply:
Modified accordingly.